pkg/repository/prisma/dbsqlc/step_runs.sql

```diff
@@ -500,14 +523,95 @@ locked_step_runs AS (
         sr."id" = @stepRunId::uuid
     FOR UPDATE SKIP LOCKED
 ),
+desired_workflow_labels AS (
+    SELECT
+        "key",
+        "strValue",
+        "intValue",
+        "required",
+        "weight",
+        "comparator"
+    FROM
+        "StepDesiredWorkerLabel"
+    WHERE
+        "stepId" = (SELECT "stepId" FROM locked_step_runs)
+),
+evaluated_affinities AS (
```

Comment:
Seems that we're calling this CTE even when there isn't a desired worker label in the step run, which will increase latency - can we case this CTE on there being a desired label?